src/OpenTracing/Util/AsyncLocalScope.cs

```diff
@@ -0,0 +1,43 @@
+namespace OpenTracing.Util
+{
+    /// <summary>
+    /// The <see cref="AsyncLocalScope"/> is a simple <see cref="IScope"/> implementation
+    /// that relies on C#'s AsyncLocal/CallContext storage primitive.
+    /// </summary>
+    /// <seealso cref="IScopeManager"/>
+    public class AsyncLocalScope : IScope
+    {
+        private readonly AsyncLocalScopeManager _scopeManager;
+        private readonly ISpan _wrappedSpan;
+        private readonly bool _finishOnDispose;
+        private readonly IScope _scopeToRestore;
+
+        public AsyncLocalScope(AsyncLocalScopeManager scopeManager, ISpan wrappedSpan, bool finishOnDispose)
+        {
+            _scopeManager = scopeManager;
+            _wrappedSpan = wrappedSpan;
+            _finishOnDispose = finishOnDispose;
+
+            _scopeToRestore = scopeManager.Active;
```

Comment:
Do we need to explicitly set the `Parent` here? I notice my InMemory implementation also does not (at this location) but I'm curious if we both missed part of the spec

Reply:
Could you clarify this? I do not understand what you mean.